lib/pull_request.rb

```diff
@@ -38,10 +38,10 @@ def validate_single_commit
   end
 
   def validate_files_changed
+    allowed_files = ["yarn.lock", "Gemfile.lock"]
     commit = GitHubClient.instance.commit("alphagov/#{@api_response.base.repo.name}", @api_response.head.sha)
     files_changed = commit.files.map(&:filename)
-    # TODO: support other package managers too (e.g. NPM)
-    files_changed == ["Gemfile.lock"]
+    (files_changed - allowed_files).empty? && (files_changed & allowed_files).any?
```

Comment:
Can you add a test for this new functionality?

I suspect the `.any?` condition isn't necessary (if only files from the `allowed_files` allowlist have been changed, there must have been at least one file changed, it wouldn't be an empty commit).

Reply:
I've made these changes now but ran into another issue with [this line](https://github.com/alphagov/govuk-dependabot-merger/blob/main/lib/change_set.rb#L45). For some reason [this PR](https://github.com/alphagov/content-publisher/pull/3120) does not have an `update-type`, not sure if it's just that one or all of the NPM PRs but will need further investigation.

Reply:
👍 suggest we convert this to Draft until that investigation is done, as this isn't ready to review until we're confident it works for all/most NPM bumps.